src/main.rs

```diff
@@ -221,20 +222,53 @@ fn main() {
     }
 }
 
+struct CommandHistory {}
+impl TypeMapKey for CommandHistory {
+    type Value = HashMap<MessageId, MessageId>;
+}
+
 struct Events {
     cmds: Commands,
 }
 
 impl RawEventHandler for Events {
     fn raw_event(&self, cx: Context, event: Event) {
         match event {
-            Event::Ready(ev) => {
+            Event::Ready(mut ev) => {
                 info!("{} connected to discord", ev.ready.user.name);
+
+                let mut cache = cx.cache.write();
+                cache.update(&mut ev);
```

Comment:
Why does the cache need this to be mutable?

Reply:
It's part of the `CacheUpdate::update` trait definition, its not used in the actual method impl at all

Reply:
Alright. That can be reviewed later, but I suspect that that's not necessary